api/v1alpha1/bucket_types.go

```diff
@@ -0,0 +1,43 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+type Bucket struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BucketSpec   `json:"spec,omitempty"`
+	Status BucketStatus `json:"status,omitempty"`
+}
+
+type BucketSpec struct {
+	// Name is the name requested for the bucket
+	Name string `json:"name"`
+	// CreationSecret is the secret that is needed to be used while creating the bucket.
```

Comment:
@shawn-hurley Would it make sense to have `provider` as a spec ? (eg: AWS/GCP/Noobaa) or add the `provider` spec as a part of `BucketBackupLocation` ?

Reply:
I think it totally makes sense to have a provider as a spec value 🤦‍♂️